app/src/main/java/com/mathewsachin/fategrandautomata/scripts/entrypoints/SupportImageMaker.kt

```diff
@@ -32,78 +30,41 @@ fun getFriendImgPath(Index: Int): File {
     return File(supportImgTempDir, "friend_${Index}.png")
 }
 
-private fun cleanExtractFolder() {
-    for (file in supportImgTempDir.listFiles()) {
-        file.delete()
-    }
-}
-
 class SupportImageMaker(private var Callback: (() -> Unit)?) : EntryPoint() {
     override fun script(): Nothing {
         initScaling()
 
         cleanExtractFolder()
 
-        val isInSupport =
-            isInSupport()
+        val isInSupport = isInSupport()
 
-        var supportBound = Region(53 * 2, 0, 143 * 2, 110 * 2)
-        val regionAnchor =
-            ImageLocator.SupportRegionTool
+        // the servant and CE images are further to the right in the friend screen
+        val supportBoundX = if (isInSupport) 53 * 2 else 88 * 2
+        var supportBound = Region(supportBoundX, 0, 143 * 2, 110 * 2)
+        val searchRegion = Region(2100, 0, 370, 1440)
 
-        val searchRegion = Region(2100, 0, 300, 1440)
-        val regionArray = searchRegion.findAll(regionAnchor, supportRegionToolSimilarity)
+        val regionAnchor = ImageLocator.SupportRegionTool
+        // At max two Servant+CE are completely on screen, so only those
+        val regionArray = searchRegion.findAll(regionAnchor, supportRegionToolSimilarity).take(2)
```

Comment:
```suggestion
        val regionArray = searchRegion.findAll(regionAnchor, supportRegionToolSimilarity)
            .take(2)
            .toArray()
```

`findAll` returns a `Sequence` instead of an `Iterable`. So, if we don't convert it into an array or list, the `count()` call below won't reuse the same computation and do it again.